binary.go

```diff
@@ -52,6 +53,7 @@ func (br *BinaryReference) Find(db *Database) *Binary {
 }
 
 // BinaryOption is the option function type for use with Binary structs
+// TODO: Remove this option and call all the KDBX version depended functions from database interface
```

Comment:
That would not really make this type obsolete. I think we'll need to keep it. The difference could be that it does not need to be exported in the future.

Reply:
Actually I think it could be replaced with bool argument of method. And it will be not necessary to store this isKDBX4 flag in each if binaries object.

Reply:
Yes, it could be.

The whole point of the function option pattern is so that we do not need to add yet another parameter whenever we come to the conclusion that it needs to be configurable for an alternative.

So the options for it are:
- add more parameters and break the function signature or keep adding specialized functions
- add an options struct and keep adding and removing fields as options (removing would actually be a breaking change) and each field needs to be optional
- use the function option pattern. This assumes that defaults are set internally and you can then pass in 0 to N options as functions which edit the struct. Specific option functions that are only part of the package can also be used to edit non exported fields

Personally I am rather fond of the third approach, because it allows for extendability without necessarily creating breaking changes

Reply:
Understood. Then in this specific case we can pass the option further to the SetContent() method and rid of the flag. Or not?